cmd/booster-http/server.go

```diff
@@ -194,6 +210,57 @@ func (s *HttpServer) handlePieceRequest(w http.ResponseWriter, r *http.Request)
 	}
 }
 
+func (s *HttpServer) handleBlockRequest(w http.ResponseWriter, r *http.Request) {
+	startTime := time.Now()
+	ctx, span := tracing.Tracer.Start(r.Context(), "http.block_cid")
+	defer span.End()
+
+	stats.Record(ctx, metrics.HttpBlockByCidRequestCount.M(1))
+
+	// Remove the path up to the payload cid
+	prefixLen := len(s.blockBasePath())
+	if len(r.URL.Path) <= prefixLen {
+		msg := fmt.Sprintf("path '%s' is missing block CID", r.URL.Path)
+		writeError(w, r, http.StatusBadRequest, msg)
+		stats.Record(ctx, metrics.HttpPayloadByCid400ResponseCount.M(1))
+		return
+	}
+
+	blockCidStr := r.URL.Path[prefixLen:]
+	blockCid, err := cid.Parse(blockCidStr)
+	if err != nil {
+		msg := fmt.Sprintf("parsing payload CID '%s': %s", blockCidStr, err.Error())
+		writeError(w, r, http.StatusBadRequest, msg)
+		stats.Record(ctx, metrics.HttpPayloadByCid400ResponseCount.M(1))
+		return
+	}
+
+	data, err := s.api.GetBlockByCid(ctx, blockCid)
+	if err != nil {
+		msg := fmt.Sprintf("server error getting data for block '%s': %s", blockCidStr, err.Error())
+		writeError(w, r, http.StatusInternalServerError, msg)
+		stats.Record(ctx, metrics.HttpBlockByCid500ResponseCount.M(1))
+		return
+	}
+
+	filename := r.URL.Query().Get("filename")
+	if filename == "" {
+		filename = blockCidStr
+	}
+
+	utf8Name := url.PathEscape(filename)
+	asciiName := url.PathEscape(regexp.MustCompile("[[:^ascii:]]").ReplaceAllLiteralString(filename, "_"))
+	w.Header().Set("Content-Disposition", fmt.Sprintf("%s; filename=\"%s\"; filename*=UTF-8''%s", "attachment", asciiName, utf8Name))
+	w.Header().Set("Content-Type", "application/vnd.ipld.raw")
```

Comment:
It looks like the Content-Type gets overridden in the parameter passed to `serveContent()`

Reply:
Fixed